pkgs/tools/misc/lorri/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, pkgs, fetchFromGitHub, rustPlatform, CoreServices, Security
+, cf-private }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "lorri";
+  version = "rolling-release-2019-10-30";
+
+  src = fetchFromGitHub {
+    owner = "target";
+    repo = pname;
+    rev = "03f10395943449b1fc5026d3386ab8c94c520ee3";
+    sha256 = "0fcl79ndaziwd8d74mk1lsijz34p2inn64b4b4am3wsyk184brzq";
+  };
+
+  BUILD_REV_COUNT = src.revCount or 1;
+  RUN_TIME_CLOSURE = pkgs.callPackage ./runtime.nix { };
+
+  nativeBuildInputs = [ pkgs.nix pkgs.direnv pkgs.which ];
+  buildInputs =
+    stdenv.lib.optionals stdenv.isDarwin [ CoreServices Security cf-private ];
+
+  cargoSha256 = "1daff4plh7hwclfp21hkx4fiflh9r80y2c7k2sd3zm4lmpy0jpfz";
+
+  # Note that https://travis-ci.org/target/lorri/builds/605036891 passed.
+  doCheck = false;
```

Comment:
Are the tests failing? We might want to enable them, because we pin a different `nixpkgs` version in the lorri repo (which we don’t update that often usually).

Reply:
Yes, they were failing due to path issues. I'll look into making them pass.

Reply:
Okay, here's what I've tried:
- If I just set `doCheck = true`, two tests fail. One of them because it fails to write something to `/nix/var`. I believe that this error does not occur upstream because of the `bogus-nixpkgs` NIX_PATH: https://github.com/target/lorri/blob/95c3e3da1be4f0d8ddb62dce21afa02d8cecfdf7/default.nix#L18. I don't really understand the effect of setting this.
- The `bogus-nixpkgs` thing in the upstream `default.nix` requires that directory to exist, and possibly also to be populated with packages that are used for testing: https://github.com/target/lorri/tree/95c3e3da1be4f0d8ddb62dce21afa02d8cecfdf7/nix/bogus-nixpkgs. I think there is a way to make this work, which is essentially "copy every .nix file from the `lorri` repo into `nixpkgs` in this PR".
- As a last alternative, I kept `doCheck = false` and tried to create an integration test in `nixpkgs/nixos/tests`. I didn't get very far unfortunately: making the Python test helpers do what I want is non-trivial.

Reply:
> * I don't really understand the effect of setting this.

Setting it in the `checkPhase` should be fine and fix that issue.


> * which is essentially "copy every .nix file from the `lorri` repo into `nixpkgs` in this PR".

Just the stuff under `nix/bogus-nixpkgs` should be necessary.

Reply:
I couldn't get it to work, and I'm not convinced it is worth cluttering the nixpkgs repo with lorri's nix files and slowing down the nixpkgs build given that `lorri` has CI. You can see my attempts in the commit history of this branch.

Reply:
@curiousleo you can just point the NIX_PATH to $src. No need to have these nix files locally.

Reply:
Yep, you can do something like

```
preCheck = ''
  export NIX_PATH=nixpkgs=${src}/nix/bogus-nixpkgs
''

Reply:
> you can just point the NIX_PATH to $src. No need to have these nix files locally.

I must have misunderstood @Profpatsch - I thought you said that there were checks preventing the use of this technique in order to ensure all nix files are actually checked into this repo.

That makes things much neater, of course! :)

Reply:
> Yep, you can do something like
> 
> ```
> preCheck = ''
>   export NIX_PATH=nixpkgs=${src}/nix/bogus-nixpkgs
> ''
> ```

I've tried this suggestion and

```nix
{
  # ...
  preCheck = ''
    export NIX_PATH="nixpkgs=${src}/nix/bogus-nixpkgs" 
  '';
}
```

The test errors with "can't write to /nix/store" persist. Do you think running the unit tests is necessary if there is an integration test (https://github.com/NixOS/nixpkgs/blob/115d1ec64f851df9db365ad7b6bf7b3e629f32d0/nixos/tests/lorri/default.nix)? If yes, I'm going to need help with getting `doCheck = true` to work.

Reply:
Let’s just set it to `false` for now, to get it in. We are running the tests upstream anyway.